bullet-journal/src/Components/Journal/VerticalLayout/Bullet.js

```diff
@@ -1,14 +1,93 @@
-import React, {useState, Fragment} from 'react'
+import React, {useState, useContext} from 'react'
+import {DayBulletsContext} from './DayView'
+import {WeekBulletsContext} from './WeekView'
 
 function Bullet(props) {
 
-    const [text, setText] = useState(props.bullet.text);
+    const [options, setOptions] = useState(false);
+    const [state, setState] = useState(
+        {
+            id: props.bullet.id,
+            date: props.bullet.date,
+            category: props.bullet.category, 
+            text: props.bullet.text,
+            completed: props.bullet.completed
+        }
+    )
+    const currBulletsContext = props.view === 'day' ? 
+        DayBulletsContext : WeekBulletsContext
+    const bulletsContext = useContext(currBulletsContext)
+    //console.log(bulletsContext)
+
+    const bulletOptionStyle = {
+        position: "fixed"
+        ,display:"block"   
+        ,zIndex:"1"                    
+        ,padding: "2px 3px 3px 3px"
+        ,transform: "translate(400%, -70%)"
+        ,border: "solid 2px #e8e3e3"
+        ,borderRadius: "5px"
+        ,backgroundColor: "white"
+    }
+    
+    const bulletTextStyle = {
+        display: "inline-block" 
+        ,padding: "0px 10px 0px 3px"
+        ,border: "0px"
+        ,outline: "none"
+        ,backgroundColor: options ? "#faf5fa" : "transparent"
+        ,textDecoration: state.completed ? "line-through" : null
+    }
 
     return (
         <div>
-            <textarea onChange={e => setText(e.target.value)}>
-                {text}
-            </textarea>
+            
+            <div style={{
+                borderWidth: "5px", 
+                padding:"3px 5px 3px 5px",
+                backgroundColor: options ? "#faf5fa" : "transparent"
+            }}
+                onMouseEnter={() => setOptions(true)}
+                onMouseLeave={() => setOptions(false)}
+            >      
+                {options ? 
+                    <div 
+                        style={bulletOptionStyle}
+                    >
+                        <span style={{left:"0px"}}>&#9200;</span>
+                        <span style={{right:"0px"}}
+                            onClick={() => bulletsContext.dispatch(
```

Comment:
Is there a `.dispatch` on context? I'm not familiar with it but a quick skim of react docs doesn't mention using dispatch. If you find it can you link me?